core/src/main/java/org/bitcoinj/base/BitcoinNetwork.java

```diff
@@ -121,6 +121,15 @@ public Coin maxMoney() {
         return MAX_MONEY;
     }
 
+    @Override
```

Comment:
I'm thinking of other subclasses of `Monetary` that are not `Fiat` but also not subclasses of `Coin`.

It would be an `IllegalArgumentException` to use anything that is not `Coin` with `BitcoinNetwork`.

Reply:
Hu, I never thought about other non-Coin monetary amounts than "Fiat". What could this possibly be? And do we really want to be prepared for that case?

On the other hand, I still don't know how an AltCoin would work, given that code like the proposed (via this PR) code seems to make it difficult.

Reply:
> What could this possibly be?

Answering my own question, precious metals comes to my mind. Or non-crypto virtual coins like Linden dollars, Fortnite V-Bucks, etc.

Reply:
Another possibility would be if Bitcoin ever uses smaller than 10^-8 units (as lightning does), an implementation using JavaMoney (JSR-354), etc.

Reply:
I'm also happy to keep things simple and use `Coin` everywhere for crypto balances. I don't have a strong feeling either way on this one.